src/MongoDB.Bson/Serialization/Attributes/BsonGuidRepresentationAttribute.cs

```
@@ -50,13 +50,15 @@ public GuidRepresentation GuidRepresentation
         /// <inheritdoc/>
         public void Apply(BsonMemberMap memberMap)
         {
-            var guidSerializer = memberMap.GetSerializer() as GuidSerializer;
-            if (guidSerializer == null)
+            var reconfiguredSerializer = SerializerConfigurator.ReconfigureSerializerRecursively(memberMap.GetSerializer(), Reconfigure);
```

Rename variable to `reconfiguredSerializer` (it's missing the letter `d`)

---

Maybe refactor this to match `BsonDateOnlyOptionsAttribute`? 

By which I mean let `base.Apply` do the recursive application and only check for `is GuidSerializer` here.

Is ``ReconfigureSerializerRecursively` even necessary any more if move the logic to`base.Apply`?

---

I guess one question might be whether we need to throw an exception when this attribute is applied to non-Guid serializers or whether that is a no-op.

---

Regarding the first question, let's discuss on slack as I fear it would be cumbersome to go back on forth on comments here 😁 

Regarding the exception, I think we should throw. I think it would be better to throw, so developers can easily see that they've applied the attribute to the wrong property.